### ISSUE TITLE
[question] Is there a way to get verbose lockfile package resolution information in the failure case?

### ISSUE THREAD

**Issue by michaelmaguire** [conan-io/conan#8475]
We are following through the examples in https://docs.conan.io/en/latest/versioning/lockfiles/ci.html#package-pipeline and we've gotten to the point where we are performing full validation builds of our downstream dependencies as indicated at the end of https://github.com/conan-io/conan/issues/8205#issuecomment-776245056.

However, sometimes our newly enacted validation fails (which it should from time-to-time if validating properly!), and I wish the failure mechanism showed more intermediate diagnostics.  e.g. here is a log of a failure:

```
buildall_verification after conan create
[Pipeline] runConanCommand
[Conan_pipeline] $ cmd.exe /C "conan lock create --reference=nxgroups/9966@PORT/stable --lockfile=export_recipe_with_full_reference.lock --lockfile-out=distribution_base_recipe_information_only.lock --base --build=missing && exit %%ERRORLEVEL%%"
nxgroups/9966@PORT/stable: Not found in local cache, looking in remotes...
nxgroups/9966@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
nxgroups/9966@PORT/stable: Downloaded recipe revision b9ad8bdeb02fdecc1acf4b2aac2f978e
By default, attempting to version this package from the Jenkins build number
conan-nxgroups get_version() os.environ['BUILD_NUMBER'][9974]
self.settings.compiler.toolset
<class 'conans.model.settings.SettingsItem'>
v140
boost/1.69.0@PORT/stable: Not found in local cache, looking in remotes...
boost/1.69.0@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
boost/1.69.0@PORT/stable: Downloaded recipe revision 1cfe498655467f7c9cad34988a94301d
zlib/1.2.11@PORT/stable: Not found in local cache, looking in remotes...
zlib/1.2.11@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
zlib/1.2.11@PORT/stable: Downloaded recipe revision 014807e02aa5748b7c315c1b7f840c1b
bzip2/1.0.6@PORT/stable: Not found in local cache, looking in remotes...
bzip2/1.0.6@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
bzip2/1.0.6@PORT/stable: Downloaded recipe revision 3d55f55fa2c67118f36a1ed254d6e438
c-ares/1.14.0@PORT/stable: Not found in local cache, looking in remotes...
c-ares/1.14.0@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
c-ares/1.14.0@PORT/stable: Downloaded recipe revision 8c8acd3f24321f451a617d2bb89fdffa
expat/2.2.5@PORT/stable: Not found in local cache, looking in remotes...
expat/2.2.5@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
expat/2.2.5@PORT/stable: Downloaded recipe revision 79818539d1b6df7dd08763c9ea3a9a7d
libcurl/7.61.1@PORT/stable: Not found in local cache, looking in remotes...
libcurl/7.61.1@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
libcurl/7.61.1@PORT/stable: Downloaded recipe revision 480f65cf8e5f15004c220a0ca16ebf2d
snappy/1.1.7@PORT/stable: Not found in local cache, looking in remotes...
snappy/1.1.7@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
snappy/1.1.7@PORT/stable: Downloaded recipe revision 82f51221f427d282eb11936accdcaa13
xercesc/3.1.1@PORT/stable: Not found in local cache, looking in remotes...
xercesc/3.1.1@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
xercesc/3.1.1@PORT/stable: Downloaded recipe revision 0
avro-cpp/1.9.1.2@PORT/stable: Not found in local cache, looking in remotes...
avro-cpp/1.9.1.2@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
avro-cpp/1.9.1.2@PORT/stable: Downloaded recipe revision 369ab06b9d88e52a7eff5f7c40c8ef61
conan_shared_python_requires/0.0.7@PORT/stable: Not found in local cache, looking in remotes...
conan_shared_python_requires/0.0.7@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
conan_shared_python_requires/0.0.7@PORT/stable: Downloaded recipe revision 0
Loaded debian\changelog to parse version[1.9.1.2] commitAtVersion[fdb224c]
basplugin/1.0.0.19@PORT/stable: Not found in local cache, looking in remotes...
basplugin/1.0.0.19@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
basplugin/1.0.0.19@PORT/stable: Downloaded recipe revision 7d8842964a5075cae0052254628dd5f1
conan_shared_python_requires/0.0.5@PORT/stable: Not found in local cache, looking in remotes...
conan_shared_python_requires/0.0.5@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
conan_shared_python_requires/0.0.5@PORT/stable: Downloaded recipe revision 0
version[1.0.0.19] commitAtVersion[master]
cfa/5.5.20@PORT/stable: Not found in local cache, looking in remotes...
cfa/5.5.20@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
fatal: bad revision '366f718..HEAD'
fatal: bad revision '31d735d..HEAD'
fatal: bad revision '57aec80..HEAD'
boost/1.69.0@PORT/stable: WARN: Setting 'cppstd' is deprecated in favor of 'compiler.cppstd', please update your recipe.
bzip2/1.0.6@PORT/stable: WARN: config() has been deprecated. Use config_options() and configure()
WARN: avro-cpp/1.9.1.2@PORT/stable: requirement boost/[>=1.60.0]@PORT/stable overridden by nxgroups/9966@PORT/stable to boost/1.69.0@PORT/stable 
WARN: avro-cpp/1.9.1.2@PORT/stable: requirement snappy/[>=1.1.7]@PORT/stable overridden by nxgroups/9966@PORT/stable to snappy/1.1.7@PORT/stable 
WARN: basplugin/1.0.0.19@PORT/stable: requirement xercesc/[>=3.1.1]@PORT/stable overridden by nxgroups/9966@PORT/stable to xercesc/3.1.1@PORT/stable 
ERROR: Conflict in dtldatatapjson/0.0.21@PORT/stable:
    'dtldatatapjson/0.0.21@PORT/stable' requires 'dtldatatap/0.0.27@PORT/stable' while 'nxgroups/9966@PORT/stable' requires 'dtldatatap/0.0.28@PORT/stable'.
    To fix this conflict you need to override the package 'dtldatatap' in your root package.
Downloading conanfile.py
Downloading conan_export.tgz
cfa/5.5.20@PORT/stable: Downloaded recipe revision ea11151147cbb86058ee2fffbdb65958
Loaded debian\changelog to parse version[5.5.20] commitAtVersion[1fed3625]
dtldatatap/0.0.28@PORT/stable: Not found in local cache, looking in remotes...
dtldatatap/0.0.28@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
dtldatatap/0.0.28@PORT/stable: Downloaded recipe revision e57eef142825efb9c88782530a1ec6fd
[dtldatatap] Loaded debian\changelog to parse version[0.0.28] commitAtVersion[366f718]
[dtldatatap] Loaded commit [366f718] after considering changelog commits
version[0.0.28] commitAtVersion[366f718]
dtldatatapjson/0.0.21@PORT/stable: Not found in local cache, looking in remotes...
dtldatatapjson/0.0.21@PORT/stable: Trying with '66c0e262-1a39-4887-be6b-7c0523ee3887'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
dtldatatapjson/0.0.21@PORT/stable: Downloaded recipe revision 411e811acfb6dc4d925e63e8fb99c87d
[dtldatatapjson] Loaded debian\changelog to parse version[0.0.21] commitAtVersion[31d735d]
[dtldatatapjson] Loaded commit [31d735d] after considering changelog commits
version[0.0.21] commitAtVersion[31d735d]
[dtldatatap] Loaded debian\changelog to parse version[0.0.27] commitAtVersion[57aec80]
[dtldatatap] Loaded commit [57aec80] after considering changelog commits
version[0.0.27] commitAtVersion[57aec80]
```

Sadly since there was a failure, no lockfile or other intermediate document seems to have been created.

It would be nice if near the *start* of the output we more info dumped such as :

```
Version ranges solved
    Version range '>=0.0.8' required by 'python_require' resolved to 'conan_shared_python_requires/0.0.9@PORT/stable' in local cache
    Version range '>=0.4' required by 'antlr3c-alloc/0.0.5@PORT/stable' resolved to 'DebuggerVisualizers/0.4@PORT/stable' in local cache
    Version range '>=1.8.0' required by 'conanfile.py (dtl-standalone/2.58.0@PORT/stable)' resolved to 'gtest/1.8.0@PORT/stable' in local cache
    Version range '>=1.6.1' required by 'conanfile.py (dtl-standalone/2.58.0@PORT/stable)' resolved to 'cmake-import-pkgconfig-targets/1.7.0@PORT/stable' in local cache
    Version range '>=1.4.0' required by 'conanfile.py (dtl-standalone/2.58.0@PORT/stable)' resolved to 'cmake-configure-bb-target/1.4.0@PORT/stable' in local cache
    Version range '>=1.12.6' required by 'conanfile.py (dtl-standalone/2.58.0@PORT/stable)' resolved to 'cmake-bb-helpers/1.13.0@PORT/stable' in local cache
    Version range '>=0.6.0' required by 'conanfile.py (dtl-standalone/2.58.0@PORT/stable)' resolved to 'cmake-bde-helpers/0.10.0@PORT/stable' in local cache
    Version range '>=0.2.0' required by 'cmake-bde-helpers/0.10.0@PORT/stable' resolved to 'cmake-module-test-helpers/0.2.0@PORT/stable' in local cache

Requirements
    DebuggerVisualizers/0.4@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    antlr3c/3.5.2-13@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    antlr3c-alloc/0.0.5@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    bde-monolith/3.42.0.4@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    dtldatatap/0.0.27@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
Python requires
    conan_shared_python_requires/0.0.2@PORT/stable
    conan_shared_python_requires/0.0.8@PORT/stable
    conan_shared_python_requires/0.0.9@PORT/stable
Packages
    DebuggerVisualizers/0.4@PORT/stable:ad8cff35a7b76d8f3d31d5bacdcfc3516c48b2fc - Download
    antlr3c/3.5.2-13@PORT/stable:ea09aaa56d61bb00e01def0e5b812c7148375fb5 - Download
    antlr3c-alloc/0.0.5@PORT/stable:37f955d190d67beed612c1bd4fe6b867e957e8d8 - Download
    bde-monolith/3.42.0.4@PORT/stable:c32596dcd26b8c708dc3d19cb73738d2b48f12a8 - Download
    dtldatatap/0.0.27@PORT/stable:d497e388c39dd03b1a1597d7ace516e3fc01ba0b - Download
Build requirements
    cmake/3.19.3@bloomberg/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    cmake-bb-helpers/1.13.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    cmake-bde-helpers/0.10.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    cmake-configure-bb-target/1.4.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    cmake-import-pkgconfig-targets/1.7.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    cmake-module-test-helpers/0.2.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    gtest/1.8.0@PORT/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
    strawberryperl/5.30.0.1@bloomberg/stable from '66c0e262-1a39-4887-be6b-7c0523ee3887' - Cache
Build requirements packages
    cmake/3.19.3@bloomberg/stable:01edd76db8e16db9b38c3cca44ec466a9444c388 - Download
    cmake-bb-helpers/1.13.0@PORT/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    cmake-bde-helpers/0.10.0@PORT/stable:4bd22f8292fc808c556e2440c204491f154100b0 - Download
    cmake-configure-bb-target/1.4.0@PORT/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    cmake-import-pkgconfig-targets/1.7.0@PORT/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    cmake-module-test-helpers/0.2.0@PORT/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    gtest/1.8.0@PORT/stable:313b5dd2de87fb01249985b40f99b7316375ef39 - Download
    strawberryperl/5.30.0.1@bloomberg/stable:456f15897172eef340fcbac8a70811f2beb26a93 - Download
```

if this were logged early on even in the case of failure, we could better debug the failure.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
I think this is a valid feature request, the ability to output information of a graph, even if the process fails because some logic error like version conflict or missing binaries.

**Comment by memsharded** [comment]
This has been improved in 2.0 already: the ``conan graph info`` information is able to output the full graph information computed until the conflict is reached. This includes full json and html graph formatted outputs, which could be useful for understanding the conflict. Note that the lockfile file is not the one containing the conflict information, because actually lockfiles have been greatly simplified in 2.0, they are no longer a graph, but a plain list of versions.

Closing the issue as solved in 2.0 then.
